core/pkg/sync/kubernetes/kubernetes_sync_test.go

```diff
@@ -443,18 +452,13 @@ func TestSync_fetch(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			// Setup with args
 			k := &Sync{
-				informer: &MockInformer{
-					fakeStore: cache.FakeCustomStore{
-						GetByKeyFunc: tt.args.InformerGetFunc,
-					},
-				},
-				readClient: &MockClient{
-					getResponse: tt.args.ClientResponse,
-					clientErr:   tt.args.ClientError,
-				},
 				logger: logger.NewLogger(nil, false),
 			}
 
+			if tt.injectionFunc != nil {
+				tt.injectionFunc(k)
```

Comment:
I like this functional test/mock setup pattern. I ended up doing it in a few of my tests a while back and was worried it was "not very Go", but it's reassuring to see you've used something similar.